commands/pipelines/diff.js

```diff
@@ -0,0 +1,157 @@
+  for (let i = githubDiff.commits.length - 1; i >= 0; i--) {
+    let commit = githubDiff.commits[i];
+    let abbreviatedHash = commit.sha.substring(0, 7);
+    let authoredDate = commit.commit.author.date;
+    let authorName = commit.commit.author.name;
+    let message = commit.commit.message.split('\n')[0];
+    console.log(`  ${abbreviatedHash}  ${authoredDate}  ${message} (${authorName})`);
+  }
+}
+
+function* fetchPipelineCoupling(heroku, appName) {
+  try {
+    return yield heroku.request({
+      method: 'GET',
+      path: `/apps/${appName}/pipeline-couplings`,
+      headers: { 'Accept': PIPELINES_HEADER }
+    });
+  } catch (err) {
+    console.log(err);
+    throw new Error(`This app (${appName}) does not seem to be a part of any pipeline.`);
+  }
+}
+
+module.exports = {
+  topic: 'pipelines',
+  command: 'diff',
+  description: 'compares the latest release of this app its downstream app(s)',
+  needsAuth: true,
+  needsApp: true,
+  run: cli.command(function* (context, heroku) {
+    const targetAppName = context.app;
+    const coupling = yield fetchPipelineCoupling(heroku, targetAppName);
+    const targetAppId = coupling.app.id;
+
+    const allApps = yield cli.action(`Fetching apps from pipeline`,
+      heroku.request({
+        method: 'GET',
+        path: `/pipelines/${coupling.pipeline.id}/apps`,
+        headers: { 'Accept': PIPELINES_HEADER }
+      }));
+
+    const sourceStage = coupling.stage;
+    const downstreamStage = PROMOTION_ORDER[PROMOTION_ORDER.indexOf(sourceStage) + 1];
+    if (downstreamStage === null || PROMOTION_ORDER.indexOf(sourceStage) < 0) {
+      throw new Error(`Unable to diff ${targetAppName}`);
+    }
+    const downstreamApps = allApps.filter(function(app) {
+      return app.coupling.stage === downstreamStage;
+    });
+
+    if (downstreamApps.length < 1) {
+      throw new Error(`Cannot diff ${targetAppName} as there are no downstream apps configured`);
+    }
+
+    // Fetch the hash of the latest release for {target, downstream[0], .., downstream[n]} apps.
+    const wrappedGetLatestCommitHash = co.wrap(_.partial(getLatestCommitHash, heroku));
+    const targetHash = yield cli.action(`Fetching latest app release for target app`,
+      wrappedGetLatestCommitHash(targetAppName, targetAppId));
+    const downstreamHashes = yield cli.action(`Fetching latest app releases for downstream apps`,
+        bluebird.all(downstreamApps.map(function (app) {
+          return wrappedGetLatestCommitHash(app.name, app.id);
+        })));
+
+    // Try to refrain from doing any GitHub/kolkrabbi API requests if none of the downstream
+    // hashes differ from the target hash.
+    const uniqueDownstreamHashes = _.uniq(_.pluck(downstreamHashes, 'hash'));
+    if (uniqueDownstreamHashes.length === 1 && uniqueDownstreamHashes[0] === targetHash.hash) {
+      console.log(`\nEverything is up to date.`);
```

Comment:
I think it would be good to print out which apps were compared here. This will give the user a little more feedback in the case that some of the apps they tried to compare are not connected to GitHub. How about:

```
my-app-staging is up to date with my-app-production
my-app-staging was not compared to my-app-production as my-app-production is not connected to GitHub
```

If this is tricky to implement for some reason then a simpler message is also fine but I think we should make the GitHub connection limitation clear.